frontend/originaltemplate.html

```diff
@@ -0,0 +1,78 @@
+                            </ul>
+                        </div>
+
+                        <!-- World Dropdown -->
+                        <button id="dropdownWorldButton" data-dropdown-toggle="dropdownWorld" class="lg:px-[65px] px-5 py-2 bg-gray-300 rounded hover:scale-105 transition transform duration-300 hover:bg-gray-200 transition transform duration-300">World</button>
+                        <div id="dropdownWorld" class="z-10 hidden bg-white divide-y divide-gray-100 rounded-lg shadow w-44">
+                            <ul class="py-2 text-sm text-gray-700">
+                                <li><a href="#" class="block px-4 py-2 hover:bg-gray-900 hover:text-white">Politics</a></li>
+                                <li><a href="#" class="block px-4 py-2 hover:bg-gray-900 hover:text-white">Conflicts</a></li>
+                                <li><a href="#" class="block px-4 py-2 hover:bg-gray-900 hover:text-white">Protests</a></li>
+                            </ul>
+                        </div>
+
+                        <!-- Search Bar -->
+                        <form class="max-w-md mx-auto w-1/3 hidden sm:block">   
+                            <label for="default-search" class="mb-2 text-sm font-medium text-gray-900 sr-only dark:text-white">Search</label>
+                            <div class="relative">
+                                <div class="absolute inset-y-0 start-0 flex items-center ps-3 pointer-events-none">
+                                    <svg class="w-4 h-4 text-gray-500 dark:text-gray-400" aria-hidden="true" xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 20 20">
+                                        <path stroke="currentColor" stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="m19 19-4-4m0-7A7 7 0 1 1 1 8a7 7 0 0 1 14 0Z"/>
+                                    </svg>
+                                </div>
+                                <input type="search" id="default-search" class="block w-full p-4 ps-10 text-sm text-gray-900 border border-gray-300 rounded-lg bg-gray-50 focus:ring-blue-500 focus:border-blue-500" placeholder="Search news..." required />
+                                <button type="submit" class="text-white absolute end-2.5 bottom-2.5 bg-blue-700 hover:bg-blue-800 focus:ring-4 focus:outline-none focus:ring-blue-300 font-medium rounded-lg text-sm px-4 py-2">Go</button>
+                            </div>
+                        </form>
+
+                    </div>
+                </div>
+            </div>
+        </div>
+    </nav>
+
+    <!-- News Grid - Dynamic Section -->
+    <div class="container mx-auto grid grid-cols-1 md:grid-cols-2 lg:grid-cols-3 gap-6 p-10 w-50" id="news-grid">
+        <!-- News articles will be injected here by JavaScript -->
+    </div>
+
+    <!-- Reference to the external JS file -->
+    <script src="./javascript/mainpage.js"></script>
```

Comment:
It is better to put "script" tags in the "head" section!

Reply:
If needed, use "onload" for "body" tag to trigger the function call in javascript file after the page is loaded.

Reply:
You're absolutely correct that placing <script> tags in the <head> section is the right convention , but when I do so,  I end up blocking the rendering of the page, which prevents the grid from coming into view.